### PR TITLE
Movie component

### PR DESCRIPTION
**Pull request by Tomislav-Busic**
I start with the creation of Movie Components

### CODE REVIEW

File: src/features/movie/components/imageContainer/ImageContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { MovieProps } from "../../../../models/movie.model";
```
Comment (dodsky) [inline src/features/movie/components/imageContainer/ImageContainer.tsx:1]: This is an import with so many levels of indirection. It makes the code fragile - if you move the component, you must ensure all these relative imports are updated. This is very error-prone.

Check if you can configure the preset like this: 
https://create-react-app.dev/docs/importing-a-component/#absolute-imports

Also, keep the component prop definitions in a component and global type definitions in the root.

File: src/features/movie/components/imageContainer/ImageContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import { AiOutlineStar } from "react-icons/ai";
+import imgIron from "../../../../assets/images/iron.jpg";
+
+import styled from "./imageContainer.module.scss";
+
+export const ImageContainer: React.FC<MovieProps> = (movie) => {
```
Comment (dodsky) [inline src/features/movie/components/imageContainer/ImageContainer.tsx:8]: Naming... naming... naming... :) 

Naming here could be better. `ImageContainer` implies a generic component, **but** it's receiving a movie as a parameter and has declared MovieProps - meaning its coupled to movie, and is some kind of movie image container. Please rename it to communicate this better.

File: src/features/movie/components/imageContainer/ImageContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <div className={styled.score}>User Score</div>
+      </div>
+      <div className={styled.details}>
+        <div className={styled.year_country}>
+          <div className={styled.year}>{year}</div>
+          <div className={styled.country}>({country})</div>
+        </div>
+        <div className={styled.genre_time}>
+          <div className={styled.genre}>
+            {genre.map((item) => item).join(", ")}
```
Comment (dodsky) [inline src/features/movie/components/imageContainer/ImageContainer.tsx:33]: Avoid doing calculations in JSX, as it's making it hard to debug. Do the `join` above and render just here, also, `map` is not doing anything, please remove it.

File: src/features/movie/components/imageContainer/imageContainer.module.scss [hunk 4]
```diff
@@ -0,0 +1,79 @@
+@import "../../../../assets/styles/variables/variables.scss";
+@import "../../../../assets/styles/mixins/mixins.scss";
```
Comment (dodsky) [inline src/features/movie/components/imageContainer/imageContainer.module.scss:2]: Same comment for imports as above.

File: src/models/movie.model.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export interface MovieProps {
```
Comment (dodsky) [inline src/models/movie.model.tsx:1]: This is a `Movie` type, not a `MovieProps` so please rename it. Props imply component params definition. You can also add an additional MovieProps which would just have Movie, e.g.
```
export interface MovieProps { 
  movie: Movie;
}
```

File: src/features/movie/components/imageContainer/ImageContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { MovieProps } from "../../../../models/movie.model";
+
+import { AiOutlineStar } from "react-icons/ai";
+import imgIron from "../../../../assets/images/iron.jpg";
```
Comment (dodsky) [inline src/features/movie/components/imageContainer/ImageContainer.tsx:4]: Will you be removing this? Avoid committing development time code, or make sure you have a TODO to remind you to remove this in the future.

**Review by dodsky — APPROVED** [review]

File: src/models/movie.model.tsx (hunk 5 above)
Comment (Tomislav-Busic) [reply src/models/movie.model.tsx:1]: Did you think I would do it like this?
export interface MovieProps { 
  movie: Movie;
}

export interface Movie {
id...
}

File: src/features/movie/components/imageContainer/ImageContainer.tsx (hunk 2 above)
Comment (Tomislav-Busic) [reply src/features/movie/components/imageContainer/ImageContainer.tsx:8]: MovieHeader?

File: src/features/movie/components/imageContainer/ImageContainer.tsx (hunk 6 above)
Comment (Tomislav-Busic) [reply src/features/movie/components/imageContainer/ImageContainer.tsx:4]: Done

File: src/features/movie/components/imageContainer/ImageContainer.tsx (hunk 3 above)
Comment (Tomislav-Busic) [reply src/features/movie/components/imageContainer/ImageContainer.tsx:33]: Done

File: src/features/movie/components/imageContainer/ImageContainer.tsx (hunk 1 above)
Comment (Tomislav-Busic) [reply src/features/movie/components/imageContainer/ImageContainer.tsx:1]: Done

File: src/features/movie/components/imageContainer/imageContainer.module.scss (hunk 4 above)
Comment (Tomislav-Busic) [reply src/features/movie/components/imageContainer/imageContainer.module.scss:2]: I don't think it works with scss, but I'm trying to find out solution.

File: src/features/movie/components/imageContainer/ImageContainer.tsx (hunk 1 above)
Comment (Tomislav-Busic) [reply src/features/movie/components/imageContainer/ImageContainer.tsx:1]: Done,

**Also, keep the component prop definitions in a component and global type definitions in the root.**

I don't fully undarstand

File: src/features/movie/components/imageContainer/imageContainer.module.scss (hunk 4 above)
Comment (Tomislav-Busic) [reply src/features/movie/components/imageContainer/imageContainer.module.scss:2]: I found a solution ->  @import "~assets/styles/variables/variables.scss";

Comment (Tomislav-Busic) [reply src/features/movie/components/imageContainer/imageContainer.module.scss:2]: I found a solution ->  @import "~assets/styles/variables/variables.scss";

File: src/models/movie.model.tsx (hunk 5 above)
Comment (dodsky) [reply src/models/movie.model.tsx:1]: Yes, but move MovieProps to the component itself.

**Pull request merged**